core/aws_ddk_core/stages/firehose_s3.py

```diff
@@ -0,0 +1,221 @@
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, List, Optional
+
+import aws_cdk.aws_kinesisfirehose_alpha as firehose
+import aws_cdk.aws_kinesisfirehose_destinations_alpha as destinations
+from aws_cdk import Duration, Size
+from aws_cdk.aws_events import EventPattern, IRuleTarget
+from aws_cdk.aws_kinesis import Stream
+from aws_cdk.aws_kms import IKey
+from aws_cdk.aws_logs import ILogGroup
+from aws_cdk.aws_s3 import IBucket
+from aws_ddk_core.pipelines.stage import DataStage
+from aws_ddk_core.resources import KinesisFactory, KinesisFirehoseFactory, S3Factory
+from constructs import Construct
+
+
+class FirehoseS3Stage(DataStage):
```

Comment:
Nitpick - `KinesisToS3` to be consistent with `SQSToLambda`? Preferring Kinesis since both a delivery and data streams could be created